src/RawRabbit/Common/NamingConventions.cs

```diff
@@ -66,13 +67,34 @@ private string GetSubscriberQueueSuffix(Type messageType)
 
 		private string GetApplicationName()
 		{
-			return Directory
-				.GetCurrentDirectory()
-				.Split('\\')
-				.Select(d => d.ToLower())
-				.Last(d => !_disallowedDirectoryNames.Contains(d))
-				.Split('.')
-				.Last();
+			return GetApplicationNameFromUrl(Directory.GetCurrentDirectory());
+		}
+
+		public string GetApplicationNameFromUrl(string url)
+		{
+			var consoleOrIISHostedAppRegex = new Regex(@".*\\(?<ApplicationName>.*[a-zA-Z][^\\]*)(\\[0-9.]*)?\\bin.*");
+			var match = consoleOrIISHostedAppRegex.Match(url);
+			var applicationName = string.Empty;
+
+			if (match.Success)
+			{
+				applicationName = match.Groups["ApplicationName"].Value;
+			}
+			else
+			{
+				var windowsServiceRegex = new Regex(@".*\\(?<ApplicationName>.*[a-zA-Z][^\\]*)(\\[0-9.]*\\)?");
+				match = windowsServiceRegex.Match(url);
+				if (match.Success)
+				{
+					applicationName = match.Groups["ApplicationName"].Value;
+				}
+			}
+
+			if (applicationName == string.Empty)
+			{
+				throw new IOException("Could not determine the application name");
```

Comment:
Don't you think it is a bit harsh to throw here? We could return `string.Empty` or perhaps `unknown`?